### ISSUE TITLE
[Inspector / Palettes] Changing Courtesy clef option often adds an extra measure to the score

### ISSUE THREAD

**Issue by Tantacrul** [musescore/MuseScore#9438]
**Describe the bug**
This is hard to describe. Please see the video:

https://user-images.githubusercontent.com/21022311/136959609-cfd318aa-7ce7-4de0-abd6-bb6cbb0a1a35.mov

Desired behaviour:

1. At the least, match the functionality of MS3.6
2. For extra points, if possible, please uncheck 'show courtesy clef', if the user removes a courtesy clef. This fixes an old MS3 problem.

**Comment by Jojo-Schmitz** [comment]
3. For extra fame: Allow (successfully) unticking that option with the courtesy clef selected.

**Comment by cbjeukendrup** [comment]
Yes, I reproduced that now. Maybe that also plays a role in causing the seemingly strange behaviour.

Another issue (probably has come up earlier): when you select a whole measure and then add a clef, the clef will be before the barline, but when you select just the measure rest, the clef will appear after the barline. 
Extra confusing: when you select a clef that is _before_ the barline and delete it, the rest in the next measure will be selected, so if you then click a clef in the palettes, that new clef will be _after_ the barline.
